gotran/codegeneration/solvercomponents.py

```diff
@@ -359,20 +382,28 @@ def __init__(
             # print expr.state.sym, expr_diff, expr_diff.args
             if expr_diff and expr.state.sym not in expr_diff.args:
 
+                linearized_name = expr.name + "_linearized"
                 linearized = self.add_intermediate(
-                    expr.name + "_linearized", expr_diff, dependent=dependent
+                    linearized_name, expr_diff, dependent=dependent
                 )
 
+                need_zero_div_check = not fraction_numerator_is_nonzero(expr_diff)
+                if not need_zero_div_check:
+                    debug("{} cannot be zero. Skipping zero division check".format(linearized_name))
+
+                RL_term = expr.sym / linearized * (sp.exp(linearized * dt) - 1)
+                if need_zero_div_check:
+                    RL_term = Conditional(
+                        abs(linearized) > delta,
+                        RL_term,
+                        dt * expr.sym,
+                    )
```

Comment:
I see, so if `linearized` is zero than you need to rewrite the RL_term as a conditional, right?

Reply:
Correct